development/cloudevents/client/client.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/go-github/v42/github"
+	"github.com/kyma-project/test-infra/development/logging"
+	console "github.com/kyma-project/test-infra/development/logging"
+	"golang.org/x/net/context"
+)
+
+// Config holds configuration for Client.
+// It can be passed to the client constructor with FromConfig client configuration option.
+type Config struct {
+	ListenPort int `envconfig:"LISTEN_PORT"`
+}
+
+// Client is a cloudevents client implementation wrapper for sending and receiving cloudevents.
+// It wraps cloudevents v2 client.
+type Client struct {
+	cloudevents.Client
+	listenPort    int
+	mux           func(cloudevents.Event)
+	eventHandlers map[string]func(*Client, cloudevents.Event)
+	Logger        logging.LoggerInterface
+}
+
+// ClientOption is a client constructor configuration option for passing configuration to the client constructor.
+type ClientOption func(*Client) error
+
+// NewClient is a constructor function for cloudevents client.
+// A constructed client can be configured by providing ClientOptions.
+// Default constructed logger is a console logger.
+// Default listen port is 8080.
+func NewClient(options ...ClientOption) (*Client, error) {
+	cc := &Client{
+		Client:        nil,
+		listenPort:    8080,
+		mux:           nil,
+		eventHandlers: nil,
+		Logger:        console.NewLogger(),
+	}
+
+	for _, opt := range options {
+		err := opt(cc)
+		if err != nil {
+			return nil, fmt.Errorf("failed applying functional option: %w", err)
```

Comment:
if we are leaving this then I'd rename the error. We don't need to explicitly say it's "functional":
```suggestion
			return nil, fmt.Errorf("failed to set option: %w", err)
```